src/mapgen/mg_biome.cpp

```diff
@@ -84,8 +84,72 @@ void BiomeManager::clear()
 	m_objects.resize(1);
 }
 
-////////////////////////////////////////////////////////////////////////////////
 
+// For BiomeGen type 'BiomeGenOriginal'
+float BiomeManager::getHeatAtPosOriginal(v3s16 pos, NoiseParams np_heat,
+	NoiseParams np_heat_blend, u64 seed)
+{
+	return
+		NoisePerlin2D(&np_heat,       pos.X, pos.Z, seed) +
+		NoisePerlin2D(&np_heat_blend, pos.X, pos.Z, seed);
+}
+
+
+// For BiomeGen type 'BiomeGenOriginal'
+float BiomeManager::getHumidityAtPosOriginal(v3s16 pos, NoiseParams np_humidity,
+	NoiseParams np_humidity_blend, u64 seed)
+{
+	return
+		NoisePerlin2D(&np_humidity,       pos.X, pos.Z, seed) +
+		NoisePerlin2D(&np_humidity_blend, pos.X, pos.Z, seed);
+}
+
+
+// For BiomeGen type 'BiomeGenOriginal'
+Biome *BiomeManager::getBiomeIdAtPosOriginal(v3s16 pos,
```

Comment:
Is there any chance to make use or merge with `BiomeGenOriginal::getBiomeAtPoint` or `BiomeGenOriginal::calcBiomeFromNoise`?
First one is unused, latter uses almost the same code.

Reply:
No, i tried, but everything in 'biomegen' is for use during mapgen so only usable in the mapgen thread only. These new functions need to be called in the server thread at any time.
`getBiomeAtPoint` is actually for use during mapgen, mgv7 used to use it years ago.